libPhoneNumber/NBPhoneNumberUtil.m

```diff
@@ -25,6 +25,17 @@
   return [aString stringByReplacingOccurrencesOfString:NB_NON_BREAKING_SPACE withString:@" "];
 }
 
+static BOOL isNan(NSString *sourceString) {
+  static dispatch_once_t onceToken;
+  static NSCharacterSet *nonDecimalCharacterSet;
+  dispatch_once(&onceToken, ^{
+    nonDecimalCharacterSet = [[NSMutableCharacterSet decimalDigitCharacterSet] invertedSet];
+  });
+
+  // Return YES if the sourceString doesn't have any characters that can be represented as a Float.
+  return !([sourceString rangeOfCharacterFromSet:nonDecimalCharacterSet].length == NSNotFound);
```

Comment:
I think you want to check for .location == NSNotFound not .length.

Did this pass unit tests? If so we should add a unit test to make it fail.

Reply:
Oops. Done.

Not sure how we can make the unit test failed in this case though, if rangeOfCharacterFromSet.location is NSNotFound, then the length is 0, which != NSNotFound (NSIntegerMax).